persistence/trees/trees.go

```diff
@@ -0,0 +1,365 @@
+	}
+	return t.getStateHash(), nil
+}
+
+func (t *treeStore) commit() error {
+	for tree := merkleTree(0); tree < numMerkleTrees; tree++ {
+		if err := t.merkleTrees[tree].Commit(); err != nil {
+			return fmt.Errorf("failed to commit %s: %w", merkleTreeToString[tree], err)
+		}
+	}
+	return nil
+}
+
+func (t *treeStore) getStateHash() string {
+	// create an order-matters list of roots
+	roots := make([][]byte, 0)
+	for tree := merkleTree(0); tree < numMerkleTrees; tree++ {
+		roots = append(roots, t.merkleTrees[tree].Root())
+	}
+
+	// combine them and hash the result
+	rootsConcat := bytes.Join(roots, []byte{})
+	stateHash := sha256.Sum256(rootsConcat)
+
+	// Convert the array to a slice and return it
+	// REF: https://stackoverflow.com/questions/28886616/convert-array-to-slice-in-go
+	return hex.EncodeToString(stateHash[:])
+}
+
+////////////////////////
+// Actor Tree Helpers //
+////////////////////////
+
+// NB: I think this needs to be done manually for all 4 types.
+func (t *treeStore) updateActorsTree(actorType coreTypes.ActorType, actors []*coreTypes.Actor) error {
+	for _, actor := range actors {
+		bzAddr, err := hex.DecodeString(actor.GetAddress())
+		if err != nil {
+			return err
+		}
+
+		actorBz, err := codec.GetCodec().Marshal(actor)
+		if err != nil {
+			return err
+		}
+
+		merkleTreeName, ok := actorTypeToMerkleTreeName[actorType]
+		if !ok {
+			return fmt.Errorf("no merkle tree found for actor type: %s", actorType)
+		}
+		if err := t.merkleTrees[merkleTreeName].Update(bzAddr, actorBz); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+//////////////////////////
+// Account Tree Helpers //
+//////////////////////////
+
+func (t *treeStore) updateAccountTrees(accounts []*coreTypes.Account) error {
+	for _, account := range accounts {
+		bzAddr, err := hex.DecodeString(account.GetAddress())
+		if err != nil {
+			return err
+		}
+
+		accBz, err := codec.GetCodec().Marshal(account)
+		if err != nil {
+			return err
+		}
+
+		if err := t.merkleTrees[accountMerkleTree].Update(bzAddr, accBz); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (t *treeStore) updatePoolTrees(pools []*coreTypes.Account) error {
+	for _, pool := range pools {
+		bzAddr, err := hex.DecodeString(pool.GetAddress())
+		if err != nil {
+			return err
+		}
+
+		accBz, err := codec.GetCodec().Marshal(pool)
+		if err != nil {
+			return err
+		}
+
+		if err := t.merkleTrees[poolMerkleTree].Update(bzAddr, accBz); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+///////////////////////
+// Data Tree Helpers //
+///////////////////////
+
+func (t *treeStore) updateTransactionsTree(indexedTxs []*coreTypes.IndexedTransaction) error {
+	for _, idxTx := range indexedTxs {
+		txBz := idxTx.GetTx()
+		txHash := crypto.SHA3Hash(txBz)
+		if err := t.merkleTrees[transactionsMerkleTree].Update(txHash, txBz); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (t *treeStore) updateParamsTree(params []*coreTypes.Param) error {
```

Comment:
<img width="866" alt="Screenshot 2023-06-19 at 13 06 04" src="https://github.com/pokt-network/pocket/assets/53987565/7c5c7c99-78ee-4507-b256-3754e10d6cd6">

Can you address these linter errors? Either bring the helper functions into the SQL package or remove them